example/mapboxExampleCamera.js

```diff
@@ -0,0 +1,284 @@
+const WORLD_SIZE = 10240000;
+const FOV_ORTHO = ( 0.1 / 180 ) * Math.PI; // Mapbox doesn't accept 0 as FOV
+const FOV = Math.atan( 3 / 4 ); // from Mapbox https://github.com/mapbox/mapbox-gl-js/blob/main/src/geo/transform.js#L93
+const EARTH_RADIUS = 6371008.8; // from Mapbox  https://github.com/mapbox/mapbox-gl-js/blob/0063cbd10a97218fb6a0f64c99bf18609b918f4c/src/geo/lng_lat.js#L11
+const EARTH_CIRCUMFERENCE_EQUATOR = 40075017; // from Mapbox https://github.com/mapbox/mapbox-gl-js/blob/0063cbd10a97218fb6a0f64c99bf18609b918f4c/src/geo/lng_lat.js#L117
+const MapConstants = {
+	WORLD_SIZE: WORLD_SIZE,
+	PROJECTION_WORLD_SIZE: WORLD_SIZE / ( EARTH_RADIUS * Math.PI * 2 ),
+	MERCATOR_A: EARTH_RADIUS,
+	DEG2RAD: Math.PI / 180,
+	RAD2DEG: 180 / Math.PI,
+	EARTH_RADIUS: EARTH_RADIUS,
+	EARTH_CIRCUMFERENCE: 2 * Math.PI * EARTH_RADIUS,
+	EARTH_CIRCUMFERENCE_EQUATOR: EARTH_CIRCUMFERENCE_EQUATOR,
+	FOV_ORTHO: FOV_ORTHO,
+	FOV: FOV,
+	FOV_DEGREES: ( FOV * 180 ) / Math.PI,
+	TILE_SIZE: 512,
+};
+
+const Constants = MapConstants;
+
+function projectToWorld( coords ) {
+
+	// Spherical mercator forward projection, re-scaling to WORLD_SIZE
+	var projected = [
+		- Constants.MERCATOR_A * Constants.DEG2RAD * coords[ 0 ] * Constants.PROJECTION_WORLD_SIZE,
+		- Constants.MERCATOR_A * Math.log( Math.tan( Math.PI * 0.25 + 0.5 * Constants.DEG2RAD * coords[ 1 ] ) ) * Constants.PROJECTION_WORLD_SIZE,
+	];
+
+	// z dimension, defaulting to 0 if not provided
+	if ( ! coords[ 2 ] ) projected.push( 0 );
+	else {
+
+		var pixelsPerMeter = projectedUnitsPerMeter( coords[ 1 ] );
+		projected.push( coords[ 2 ] * pixelsPerMeter );
+
+	}
+
+	var result = new THREE.Vector3( projected[ 0 ], projected[ 1 ], projected[ 2 ] );
+
+	return result;
+
+}
+
+function projectedUnitsPerMeter( latitude ) {
+
+	return Math.abs(
+		Constants.WORLD_SIZE / Math.cos( Constants.DEG2RAD * latitude ) / Constants.EARTH_CIRCUMFERENCE );
+
+}
+
+class CameraSync {
+
+	constructor( map, camera, world ) {
+
+		this.map = map;
+		this.camera = camera;
+		this.active = true;
+		// We're in charge of the camera now!
+		this.camera.matrixAutoUpdate = false;
+		// Postion and configure the world group so we can scale it appropriately when the camera zooms
+		this.world = world || new THREE.Group();
+		this.world.position.x = this.world.position.y =
+    MapConstants.WORLD_SIZE / 2;
+		this.world.matrixAutoUpdate = false;
+		// set up basic camera state
+		this.state = {
+			translateCenter: new THREE.Matrix4().makeTranslation(
+				MapConstants.WORLD_SIZE / 2,
+				- MapConstants.WORLD_SIZE / 2,
+				0
+			),
+			worldSizeRatio: MapConstants.TILE_SIZE / MapConstants.WORLD_SIZE,
+			worldSize: MapConstants.TILE_SIZE * this.map.transform.scale,
+		};
+		// Listen for move events from the map and update the Three.js camera
+		const _this = this;
+		this.map
+			.on( 'move', function () {
+
+				_this.updateCamera();
+
+			} )
+			.on( 'resize', function () {
+
+				_this.setupCamera();
+
+			} );
```

Comment:
If we use arrow functions there's no need to declare a local scope "_this". ie

```js
this.map
	.on( 'move', () => {

		this.updateCamera();

	} )
```

Reply:
ok